codes/extractive/netsum/netsum.py

```diff
@@ -0,0 +1,219 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Oct 26 21:41:03 2018
+
+@author: Rose
+"""
+from __future__ import division
+import nltk
+from nltk.tokenize import sent_tokenize, word_tokenize, RegexpTokenizer
+import pandas as pd
+import numpy as np
+from collections import Counter
+import h5py
+import os
+import keras
+from sklearn.model_selection import train_test_split
+from keras.models import Sequential, Model
+from keras.layers import Dense, Dropout, Activation, Flatten, Conv2D, MaxPooling2D, Input
+from keras.callbacks import ModelCheckpoint, EarlyStopping
+
+text_path = "/root/nlp_project/cnn/text_2000/"
+summary_path = "/root/nlp_project/cnn/summary_2000/"
+file_list = "/root/nlp_project/cnn/datalist2.csv"
+
+# min. input size: 30 (sentences)
+MAX_INPUT_SEQ_LENGTH=30
+BATCH_SIZE = 20
+EPOCHS = 20
+
+# load the data
+def data_prep(text_path, summary_path, file_info, save_transformed_flag, transformed_path=None):
+    label = []
+    feature= []
+    data_df = pd.read_csv(file_list,sep=',',header='infer')
+    word_tokenizer = RegexpTokenizer(r'\w+')
+    
+    for i in range(len(data_df)):
+        label_list = []
+        feature_list = []
+        padding_needed = False
+        text_file = os.path.join(text_path,data_df["text_path"][i])
+        with open(text_file, 'r', encoding="utf8") as rf:
+            text = rf.read()
+            text_sents = sent_tokenize(text)
```

Comment:
Good to know this usage. Seem we have some subtitles in text. Does this method ignore them or treat them as sentences (may not be a good idea)?

like

subtitle1
sent1. sent2.
subtitle2
sent3.
sent4.

Reply:
Yes, I think function sent_tokenize means it would transform text into a sequence of sentences

Reply:
This script assumes that the body and the summary are already separated and saved in two files. The pre-processing process has already disregard any subtitles in the body. Plus, not all texts have subtitles, so I chose not to operate on that granular level.